core/src/main/java/io/grpc/internal/Stream.java

```diff
@@ -77,4 +79,10 @@
    * result in excessive buffering within the transport.
    */
   boolean isReady();
+
+  /**
+   * Sets the default message encoder for messages on this stream.
+   * @param c the compressor / decompressor
+   */
+  void setDefaultCodec(Codec c);
```

Comment:
The codec to use by default if a message is compressed (and not used if not compressed).